### PR TITLE
[NHUB-315] (2.4.2): Support Audio & Video for Featuremedia

### PR DESCRIPTION
**Pull request by MarkLark86**
## Checklist

- [x] This pull request is not adding new forms that use redux
- [x] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence

### CODE REVIEW

File: assets/wire/utils.js [hunk 1]
```diff
@@ -49,6 +49,35 @@ function getRelatedItemsByType(item, type) {
     return item.type === type ? [item] : Object.values(get(item, 'associations', {}) || {}).filter((assoc) => get(assoc, 'type') === type);
 }
 
+export function getFeatureMedia(item) {
+    if (['picture', 'video', 'audio'].includes(item.type)) {
+        return item;
+    }
+
+    const featured = get(item, 'associations.featuremedia');
```
Comment (tomaskikutis) [inline assets/wire/utils.js:57]: it'd be nice to replace lodash get calls with optional chaining so we don't lose typescript support (find usages, rename)

File: assets/wire/utils.js [hunk 2]
```diff
@@ -49,6 +49,35 @@ function getRelatedItemsByType(item, type) {
...
+}
+
+export function getOtherMedia(item) {
+    if (['picture', 'video', 'audio'].includes(item.type)) {
+        return null;
+    }
+
+    return Object.keys(item.associations || {})
+        .filter((key) => (
+            !key.startsWith('editor_') &&
```
Comment (tomaskikutis) [inline assets/wire/utils.js:73]: what the meaning of `key.startsWith('editor_')`?

**Review by tomaskikutis — APPROVED** [review]

File: assets/wire/utils.js (hunk 1 above)
Comment (MarkLark86) [reply assets/wire/utils.js:57]: This PR is the javascript version for v2.4 of Newshub. The other PR is for typescript (v2.5+) which does not use the `locash.get` function.

File: assets/wire/utils.js (hunk 2 above)
Comment (MarkLark86) [reply assets/wire/utils.js:73]: If you look at `apps.archive.archive.update_associations` in the backend (https://github.com/superdesk/superdesk-core/blob/develop/apps/archive/archive.py#L212), media items from `fields_meta.body_html.draftjsState.body_html` metadata are stored in `assoications` under the `editor_` prefix

**Pull request merged**